django-stubs/contrib/gis/db/models/fields.pyi

```diff
@@ -108,7 +108,9 @@ class GeometryField(BaseSpatialField[_ST, _GT]):
         validators: Iterable[_ValidatorCallable] = ...,
         error_messages: _ErrorMessagesMapping | None = ...,
     ) -> None: ...
-    def formfield(self, **kwargs: Any) -> Any: ...  # type: ignore[override]
+    def formfield(  # type: ignore[override]
+        self, form_class: type[forms.GeometryField] | None = ..., geom_type: str = ..., srid: Any = ..., **kwargs: Any
```

Comment:
Since these are handled from `**kwargs`, you should typehint them as keyword-only parameters. That should get rid of the allowlist ignores too.

```suggestion
        self, *, form_class: type[forms.GeometryField] | None = ..., geom_type: str = ..., srid: Any = ..., **kwargs: Any
```

Reply:
The `*` keyword-only change only affects the `GeometryField` method, not `Field`. So it should be accurate AFAICT. (https://github.com/django/django/blob/stable/4.2.x/django/contrib/gis/db/models/fields.py#L303)